Kentico/kontent-delivery-sdk-net
# Returns back the IOptionsMonitor for options hotreloading

### Motivation

Returns back the `IOptionsMonitor` that was removed in https://github.com/Kentico/kontent-delivery-sdk-net/releases/tag/14.0.0-rc3 to preserve the options hotreloading functionality.

### Checklist


### How to test

- try various ways of registering the client and the cache (`AddDeliveryClient()` and `AddDeliveryClientCache()`)
- make sure that updating options (e.g. updating appSettings.json) during the app runtime gets reflected in the app...e.g. try switching to preview (or try the setup wizard in https://github.com/Kentico/kontent-sample-app-net/)
